CMakePresets.json

```diff
@@ -0,0 +1,33 @@
+{
+  // Pass `--preset <name>` to CMake to use a preset defined here.
+  //
+  // We curently use version 3 presets which require CMake 3.21+.
+  // Unfortunately, CMake versions before 3.19 may ignore the `--preset`
+  // argument silently. So if the preset is not picked up, make sure you
+  // have a recent enough CMake version.
+  //
+  // Note: JSON does not support officially, but CMake apparently
+  // ignores lines starting with whitespace followed by "//" silently.
```

Comment:
Urghs, thanks for doing the research here. So they first enabled comment support intentionally, then disabled it intentionally.  Currently, [it just works by accident](https://gitlab.kitware.com/cmake/cmake/-/issues/21858#note_908543). 

I'm not sure how we should deal with this. Having the comments is risky because future versions may not support them. (This PR here uses presets only for dev mode, but we may add other presets, e.g., for coverage....)

It seems that CMake's parser defeats all the hacks I could imagine or find in https://stackoverflow.com/questions/244777/can-comments-be-used-in-json. Things we could do:
 - Have a template file and strip `//` lines using grep, generating the actual `.json` from it. But this needs some mechanism to make sure that the file is up to date etc... All doable but seems overkill for this tiny issue.
 - (https://hjson.github.io/ is an entire off-the-shelf solution for this, but that adds a maintainer dependency and is hard to remember.)   
 - We could just drop the comments. CMake templates support a `cmakeMinimumRequired` field which I think is enough to document the need for CMake 3.21. A better warning should then be added to build docs (to be written :D). The comment about the missing generator thing can be in the commit message (not optimal but okayish).

I lean towards dropping comments. Sad that we need to spend time on this...

Reply:
> I lean towards dropping comments.

Agree.